### PR TITLE
feat(waf): add a datasource to get the list of cc protection rules

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a datasource to get the list of cc protection rules.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support a new datasource used to get list of cc protection rules.
2. support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST="./huaweicloud/services/acceptance/waf" TESTARGS="-run TestAccDataSourceRulesCcProtection_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccDataSourceRulesCcProtection_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceRulesCcProtection_basic
=== PAUSE TestAccDataSourceRulesCcProtection_basic
=== CONT  TestAccDataSourceRulesCcProtection_basic
--- PASS: TestAccDataSourceRulesCcProtection_basic (391.30s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       391.380s
```

### CODE REVIEW

File: huaweicloud/services/waf/data_source_huaweicloud_waf_rules_cc_protection.go [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the status of the cc protection rule.`,
+			},
+			"enterprise_project_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the enterprise project ID to which the protection policies belong.`,
```
Comment (jinyangyang222) [inline huaweicloud/services/waf/data_source_huaweicloud_waf_rules_cc_protection.go:55]:
```suggestion
				Description: `Specifies the enterprise project ID to which the protection policy belongs.`,
```

Comment (ruwenqiang123) [reply huaweicloud/services/waf/data_source_huaweicloud_waf_rules_cc_protection.go]: done

**Pull request merged**